model_hub/docker/publish-docker.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+set -e
+set -x
+
+docker info
+
+if [ "$#" -ne 5 ] ; then
+    echo "usage: $0 LOG_NAME BASE_TAG HASH VERSION ARTIFACTS_DIR" >&2
+    exit 1
+fi
+
+log_name="$1"
+base_tag="$2"
+hash="$3"
+version="$4"
+artifacts="$5"
+
+underscore_name="$(echo -n "$log_name" | tr - _)"
+
+docker push "$base_tag-$hash"
+docker push "$base_tag-$version"
+
+mkdir -p "$artifacts"
+
+log_file="$artifacts/publish-$log_name"
+(
+    echo "${underscore_name}_hashed: $base_tag-$hash"
+    echo "${underscore_name}_versioned: $base_tag-$version"
+) > "$log_file"
```

Comment:
* for upstream branches we will build the images corresponding to the git hash and the user can optionally run model_hub e2e tests with the image.  this should be the path we take for future changes to model_hub since CI will build docker images and we can run e2e tests against that image if desired.
* for master branch, when commits are merged we will build the corresponding image and tag that with the current dev version.  this image will be used for nightly tests and for any user that clones the example from master.
* at release time, `bumpversion` will change the environment image in the configs to the new version and CI will build and publish images with that version as the tag.